src/components/PeoplePicker/Principal.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { IPrincipalProps, IPrincipal, PrincipalType } from './Principal.Props';
+import { getId } from '../../utilities/getId';
+import { autobind } from '../../utilities/autobind';
+import './Principal.scss';
+import { Icon } from './../../../src/components/Icon/Icon';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { DirectionalHint } from '../../utilities/DirectionalHint';
+
+export interface IPrincipalState {
+    showTooltip?: boolean;
+}
+
+export class Principal extends React.PureComponent<IPrincipalProps, IPrincipalState> {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showTooltip: false
+        };
+    }
+
+    @autobind
+    private _onClickPrincipal(): void {
+        this.props.onSelect(this.props.principal);
+    }
+
+    @autobind
+    private _onClickDelete(): void {
+        this.props.onDelete(this.props.principal);
+    }
+
+    @autobind
+    private _onMouseOver(): void {
+        this.setState({ showTooltip: true });
+    }
+
+    @autobind
+    private _onMouseOut(): void {
+        this.setState({ showTooltip: false });
+    }
+
+    @autobind
+    private _rednerTypeIcon(): JSX.Element {
```

Comment:
Can't you just change iconName and className based on principal type and have just one return expression?

Reply:
PrincipalType is enum, and PrincipalType.user is equal to some number. So I don't understand what do you mean?

Reply:
There is a typo in the name of this function.
It seems to me that @ivansicic is suggesting, to have the function to return iconName and className instead of this function that returns JSX element. I agree, it will be more readable.